pkg/acceptance/cluster/dockercluster.go

```diff
@@ -916,23 +896,19 @@ func (l *DockerCluster) ExecCLI(ctx context.Context, i int, cmd []string) (strin
 	return outputStream.String(), errorStream.String(), nil
 }
 
-func ensureLogDirExists(logDir string) {
-	// Ensure that the path exists, with all its parents.
-	// If we don't make sure the directory exists, Docker will and then we
-	// may run into ownership issues (think Docker running as root, but us
-	// running as a regular Joe as it happens on CircleCI).
-	maybePanic(os.MkdirAll(logDir, 0755))
-	// Now make the last component, and just the last one, writable by
-	// anyone, and set the gid and uid bit so that the owner is
-	// propagated to sub-directories.
-	maybePanic(os.Chmod(logDir, 0777|os.ModeSetuid|os.ModeSetgid))
-}
-
-// Cleanup removes the log directory if it was initially created
-// by this DockerCluster.
-func (l *DockerCluster) Cleanup(ctx context.Context) {
-	if l.logDir != "" && l.logDirRemovable {
-		if err := os.RemoveAll(l.logDir); err != nil {
+// Cleanup removes the cluster's volumes directory, optionally preserving the
+// logs directory.
+func (l *DockerCluster) Cleanup(ctx context.Context, preserveLogs bool) {
+	volumes, err := ioutil.ReadDir(l.volumesDir)
+	if err != nil {
+		log.Warning(ctx, err)
+		return
+	}
+	for _, v := range volumes {
+		if preserveLogs && v.Name() == "logs" {
+			continue
+		}
+		if err := os.RemoveAll(filepath.Join(l.volumesDir, v.Name())); err != nil {
```

Comment:
Makes me nervous, because I tend to run with `-l .` which I think might translate to `volumesDir = $GOPATH/src/github.com/cockroachdb/cockroach/acceptance` and you'd just maul my checkout.

Reply:
Guess in practice it's fine because `volumesDir` is either a tempdir or my originally specified directory with the clusterID joined, right? Perhaps add a comment on `volumesDir`.